src/credentials/providers/ssoCredentialProvider.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { Credentials, SSO } from 'aws-sdk'
+import { getLogger } from '../../shared/logger'
+import { SsoAccessTokenProvider } from '../sso/ssoAccessTokenProvider'
+
+export class SsoCredentialProvider {
+    private ssoAccount: string
+    private ssoRole: string
+    private ssoClient: SSO
+    private ssoAccessTokenProvider: SsoAccessTokenProvider
+
+    constructor(ssoAccount: string, ssoRole: string, ssoClient: SSO, ssoAccessTokenProvider: SsoAccessTokenProvider) {
```

Comment:
Nit: mark constructors as `public` and reduce lines by removing the field declarations and consolidating into the constructor (also applies elsewhere):
```
    public constructor(
        private ssoAccount: string,
        private ssoRole: string,
        private ssoClient: SSO,
        private ssoAccessTokenProvider: SsoAccessTokenProvider
    ) { }
```

Reply:
Applied to all classes.  Much better, thanks.